main/saturn-node.js

```diff
@@ -4,6 +4,7 @@ const path = require('path')
 const fs = require('node:fs/promises')
 const { setTimeout } = require('timers/promises')
 const assert = require('node:assert')
+const { request } = require('undici')
```

Comment:
could we simplify the backend by also using fetch, the JS standard api?

Reply:
We can use fetch if we are ok with the ramifications:
- Fetch is not a built-in API in Node.js 16. We will keep using Undici, but with a different API (`fetch` vs `request`).
- The performance of the `fetch` API in both Undici and Node.js 18 is much worse than Undici's `request` API due to WebStreams. Our current request/fetch API usage is not performance critical; thus, it's ok to prefer DX over performance.

Reply:
+1 to using fetch via undici

Do you know how much slower fetch is?

Reply:
I am not sure. If I am reading this comment from Feb 2022 correctly, then `fetch` was 8x slower than `request`.
https://github.com/nodejs/undici/issues/1203#issuecomment-1033189710

The Undici team has landed several performance improvements since then, but the real problem is in WebStreams. Quoting from https://github.com/nodejs/undici/issues/1203#issuecomment-1100969210:
> However, the major bottleneck is under WebStreams implementation in Node.js core, thus, I don't think there is a clear path for undici.fetch to improve its performance significantly without touching Node.js internals.

Reply:
Done in https://github.com/filecoin-project/filecoin-station/pull/136/commits/66cab6f94a376971d952e936a010fa34a3e8b309.

I find the `request` API slightly more ergonomic than the `fetch` API, but I guess there is more value in using a standard Web API that more people are familiar with.